### ISSUE TITLE
Recreate winning kaggle solution: train

### ISSUE THREAD

**Issue by calebchin** [calebchin/BELKAml#4]
Take winning kaggle solution and re-implement it in pytorch and not in notebook. This issue only covers the loss portion.

Solution: https://www.kaggle.com/code/victorshlepov/belka2024-1stplacesolution/notebook
